mapie/classification.py

```diff
@@ -684,9 +680,11 @@ def fit(
         X, y = indexable(X, y)
         y = _check_y(y)
         assert type_of_target(y) == "multiclass"
-        self.n_classes_ = len(set(y))
-        self.n_features_in_ = check_n_features_in(X, cv, estimator)
         sample_weight, X, y = check_null_weight(sample_weight, X, y)
+        y = cast(NDArray, y)
```

Comment:
`y` is casted to a `NDArray` but `y` accepted by `_fit_and_predict_oof_model` is still an `ArrayLike`.  I have a related open question : shouldn't we convert all input `ArrayLike` arguments as `NDArray` at the beginning of `fit` and `predict` methods so all subsequent methods only accept and return `NDArray` arguments ?

Reply:
Alas no, because as some users noticed in this issue https://github.com/scikit-learn-contrib/MAPIE/issues/128, MAPIE should not convert input arrays as numpy arrays on its own, but should let the wrapped model manage arbitrary inputs. One example is a wrapped model being a pipeline treating pandas dataframe with a scikit-learn ColumnTransformer : the latter needs input to be pandas dataframe in order to record column names. Would MAPIE convert this to a numpy array, the compatibility with such pipelines would be lost.

Reply:
You are right that `y` is converted to `np.array` as soon as `_check_y` is called, but mypy does not notice it because this scikit-learn function has no typing. The cast is then necessary, here or before `check_null_weight`. In this latter case, for mypy to understand that the return `y` is `NDArray`,  the signature of `check_null_weight` should changed.

Reply:
OK understood, thanks for the explanation. So why is `y` casted here ? Because transforming it from a `pd.Series` to a `NDArray` won't affect the scikit-learn pipeline ?

Reply:
`y` needs to be casted to `NDArray` because MAPIE code uses typical `NDArray` methods like `reshape` further in the code.